scripts/5_acbm_to_matsim_xml.py

```diff
@@ -0,0 +1,163 @@
+from acbm.cli import acbm_cli
+from acbm.config import load_config
+from acbm.logger_config import converting_to_matsim_logger as logger
+from acbm.postprocessing.matsim import (
+    add_home_location_to_individuals,
+    calculate_percentage_remaining,
+    filter_by_pid,
+    filter_no_location,
+    log_row_count,
+)
+
+
+@acbm_cli
+def main(config_file):
+    config = load_config(config_file)
+    config.init_rng()
+
+    # ----- Read the data
+
+    logger.info("1 - Loading data")
+
+    individuals = pd.read_csv(
+        acbm.root_path / "data/processed/activities_pam/people.csv"
+    )
+    households = pd.read_csv(
+        acbm.root_path / "data/processed/activities_pam/households.csv"
+    )
+    activities = pd.read_csv(
+        acbm.root_path / "data/processed/activities_pam/activities.csv"
+    )
+    legs = pd.read_csv(acbm.root_path / "data/processed/activities_pam/legs.csv")
+    legs_geo = pd.read_parquet(
+        acbm.root_path / "data/processed/activities_pam/legs_with_locations.parquet"
+    )
+
+    # ----- Clean the data
+
+    logger.info("2 - Cleaning data")
+
+    # We will be removing some rows in each planning operation. This function helps keep a
+    # record of the number of rows in each table after each operation.
+
+    row_counts = []
+
+    logger.info("2.1 - Record number of rows in each df before cleaning")
+
+    log_row_count(individuals, "individuals", "0_initial", row_counts)
+    log_row_count(households, "households", "0_initial", row_counts)
+    log_row_count(activities, "activities", "0_initial", row_counts)
+    log_row_count(legs, "legs", "0_initial", row_counts)
+    log_row_count(legs_geo, "legs_geo", "0_initial", row_counts)
+
+    logger.info("2.2 - Remove people that don't exist across all dfs")
+
+    # When writing to matsim using pam, we get an error when a pid exists in one dataset
+    #  but not in the other. We will remove these people from the datasets.
+
+    individuals, activities, legs, legs_geo, households = filter_by_pid(
+        individuals, activities, legs, legs_geo, households
+    )
+
+    log_row_count(individuals, "individuals", "1_filter_by_pid", row_counts)
+    log_row_count(households, "households", "1_filter_by_pid", row_counts)
+    log_row_count(activities, "activities", "1_filter_by_pid", row_counts)
+    log_row_count(legs, "legs", "1_filter_by_pid", row_counts)
+    log_row_count(legs_geo, "legs_geo", "1_filter_by_pid", row_counts)
+
+    logger.info("2.3 - Rename geometry columns (for PAM)")
+    # TODO: Rename columns upstream in 3.3_assign_facility_all script
+    legs_geo.rename(
+        columns={
+            "start_location_geometry_wkt": "start_loc",
+            "end_location_geometry_wkt": "end_loc",
+        },
+        inplace=True,
+    )
+
+    logger.info("2.4 - Remove people with missing location data ")
+
+    individuals, households, activities, legs, legs_geo = filter_no_location(
+        individuals, households, activities, legs, legs_geo
```

Comment:
I think the arguments might need to be passed in the order below. Alternatively, named arguments could be good here. 
```suggestion
        individuals, activities, legs, legs_geo, households
```

Reply:
Yes I spent some time scratching my head over this error! I fixed in #73 that should be merged into #56. See https://github.com/Urban-Analytics-Technology-Platform/acbm/pull/73/commits/a1da715d8ea8724468f71d0dc3c9cbb497d5aaba and https://github.com/Urban-Analytics-Technology-Platform/acbm/pull/73/commits/18f77d86c6373d5591070044a4e7a73522fd46d0

Reply:
Great thanks! I think tomorrow we should merge #73 into this, and then merge this into main

Reply:
Sounds good!

Reply:
Resolved from merging #73